crates/iceberg-catalog/src/implementations/postgres/task_queues/mod.rs

```diff
@@ -66,26 +66,30 @@ async fn queue_task(
     let task_id = Uuid::now_v7();
     Ok(sqlx::query_scalar!(
         r#"INSERT INTO task(
-                        task_id,
-                        task_name,
-                        status,
-                        parent_task_id,
-                        idempotency_key,
-                        warehouse_id,
-                        suspend_until)
+                task_id,
+                queue_name,
+                status,
+                parent_task_id,
+                idempotency_key,
+                warehouse_id,
+                suspend_until)
         VALUES ($1, $2, 'pending', $3, $4, $5, $6)
-        ON CONFLICT ON CONSTRAINT unique_idempotency_key DO NOTHING
+        ON CONFLICT ON CONSTRAINT unique_idempotency_key
+        DO UPDATE SET
+            status = EXCLUDED.status,
+            suspend_until = EXCLUDED.suspend_until
+        WHERE task.status = 'cancelled'
```

Comment:
This is basically resume task, we could also use the same for retries. 

Alternatively, we do not update cancelled tasks but instead we create a new task if a previously restored table is to be dropped again.

Reply:
@c-thiel

Reply:
I think the way you implemented (retries) is clearer. If we had multiple tasks for a single table it's less clear how to perform the update. One task per job per table seems clearest to me.